ts/test/test-init-config.ts

```diff
@@ -369,6 +369,45 @@ describe('createProfiler', () => {
        assert.deepEqual(profiler.config, expConfig);
      });
 
+  it('should get values from Knative environment variables when values not specified in config or other environment variables',
+     async () => {
+       process.env.K_SERVICE = 'k-service';
```

Comment:
It would be good to get confirmation from @steren that K_SERVICE won't ever contain uppercase letters and, more generally, will always match the profiler's requirements for service name strings.

Reply:
K_service follows the Kubernetes resource naming requirements. As far a I understood, uppercase is not allowed.

Reply:
It looks like kubernetes might allow starting a name K_SERVICE with a number ([this kubernetes doc](https://kubernetes.io/docs/concepts/overview/working-with-objects/names/) says "By convention, the names of Kubernetes resources should be up to maximum length of 253 characters and consist of lower case alphanumeric characters, -, and ., but certain resources have more specific restrictions."), but otherwise should be safe.

I think it's probably still a worthwhile change.